tests/core/management/commands/test_send_push_notifications.py

```diff
@@ -1,33 +1,155 @@
     """
-    Ensure that disabled push notifications cause an error
+    Test that cover the send_push_notification management command and its filter mechanisms
     """
-    settings.FCM_ENABLED = False
-    with pytest.raises(CommandError) as exc_info:
-        assert not any(get_command_output("send_push_notifications"))
-    assert str(exc_info.value) == "Push notifications are disabled"
 
+    @pytest.mark.django_db
+    def test_push_notifications_disabled(self, settings: SettingsWrapper) -> None:
+        """
+        Ensure that disabled push notifications cause an error
+        """
+        settings.FCM_ENABLED = False
+        with pytest.raises(CommandError) as exc_info:
+            assert not any(get_command_output("send_push_notifications"))
+        assert str(exc_info.value) == "Push notifications are disabled"
 
-@pytest.mark.django_db
-def test_push_notifications_nonexisting_testregion(settings: SettingsWrapper) -> None:
-    """
-    Ensure that an error is caused when the system runs in debug mode but the test region does not exist
-    """
-    settings.TEST_REGION_SLUG = "non-existing"
-    settings.DEBUG = True
-    with pytest.raises(CommandError) as exc_info:
-        assert not any(get_command_output("send_push_notifications"))
-    assert (
-        str(exc_info.value)
-        == f"The system runs with DEBUG=True but the region with TEST_REGION_SLUG={settings.TEST_REGION_SLUG} does not exist."
-    )
+    @pytest.mark.django_db
+    def test_push_notifications_nonexisting_testregion(
+        self,
+        settings: SettingsWrapper,
+    ) -> None:
+        """
+        Ensure that an error is caused when the system runs in debug mode but the test region does not exist
+        """
+        settings.TEST_REGION_SLUG = "non-existing"
+        settings.DEBUG = True
+        with pytest.raises(CommandError) as exc_info:
+            assert not any(get_command_output("send_push_notifications"))
+        assert (
+            str(exc_info.value)
+            == f"The system runs with DEBUG=True but the region with TEST_REGION_SLUG={settings.TEST_REGION_SLUG} does not exist."
+        )
+
+    patch: Any = None
+
+    def setup_method(self) -> None:
+        self.patch = patch.object(
+            FirebaseApiClient, "_get_access_token", return_value="secret access token"
+        )
+        self.patch.start()
+
+    def teardown_method(self) -> None:
+        self.patch.stop()
+        self.patch = None
+
+    @pytest.mark.django_db
+    def test_ignore_overdue_notification(
+        self, settings: SettingsWrapper, requests_mock: Mocker
+    ) -> None:
+        called = 0
+
+        def json_func(
+            _request: requests.PreparedRequest, _context: Any
+        ) -> dict[str, str | int]:
+            nonlocal called
+            called += 1
+
+            return {
+                "name": "projects/integreat-2020/messages/1",
+                "status_code": 200,
+            }
+
+        requests_mock.post(
```

Comment:
Couldn't we just mock the `send_pn` method to return True?

Reply:
Maybe but I spent a significant amount of time setting up the mock and was really relieved when it worked. 😄

Reply:
:eyes: